### PR TITLE
Add expire/renewal information to plan page

### PR DESCRIPTION
**Pull request by sirreal**
This expire/renew logic is naive and does not follow the more robust logic used on the purchases page which gives "real" renewal/expiration which accounts for a number of things including payment type and credit card expiration.

Partial implementation of #7228

### CODE REVIEW

**Comment by rickybanister** [comment]
Can you explain why it's more naive or why we can't just use the same logic from /purchases?

**Comment by sirreal** [comment]
> Can you explain why it's more naive or why we can't just use the same logic from /purchases? 

@rickybanister Sure 🙂 

TL;DR We _can and should_ use the same logic, but rather than duplicate the logic we should work out how to extract it and make it available for other modules.

The purchases page deals with a purchase object and has a [significant amount of logic](https://github.com/Automattic/wp-calypso/blob/7b57ac64f77bcd764a21a649b619659800e7767e/client/me/purchases/manage-purchase/purchase-meta.jsx#L149-L183) just to decide whether it should display "Expires" or "Renews". I'm not familiar with this code, it's just something I noticed as I implemented this. It appears to check payment type, credit card expiration, and many factors that ostensibly help determine if a plan _will likely be able to be renewed_, which is different from just having `autoRenew` activated.

On the plans page, I don't have a purchase object, but something else. This implementation is naïve because I look at some obvious properties on this object like [`autoRenewDateMoment`](https://github.com/Automattic/wp-calypso/blob/7b57ac64f77bcd764a21a649b619659800e7767e/client/me/purchases/manage-purchase/purchase-meta.jsx#L149-L183) and [`userFacingExpiryMoment `](https://github.com/Automattic/wp-calypso/blob/c11ad3a458c863785defb90818c7066f8d3c5d4b/client/my-sites/plan-features/header.jsx#L90). Apart from shuffling some code around we'd need to query the `purchase` object to have the same information as on the purchases page.

That's the overview of why this is naïve and how a more robust implementation would need to change.

**Comment by sirreal** [comment]
@rickybanister Would you like me to go ahead with the more accurate "expires/renews" implementation?

**Comment by rickybanister** [comment]
Yes please. I think it's worth 'doing this right' if possible.

**Comment by sirreal** [comment]
I'm wondering if I have a reliable way to get to subscription data (expiration/renewal dates) that would match what we see in `/me/purchases/:site/:purchaseId`. I'm wondering how and if this is possible to do reliably, the difference being that here we assume we know an active plan, and in purchases we're working with a specific purchase.

I've started exploring querying site purchases, it seems the way forward may be to work with the array of site purchases and try to find a matching purchase, although it seems a bit fragile.

@ryelle do you have any input as the one who working on the `/me/purchases` side in #13093?

**Comment by rickybanister** [comment]
We should also account for the post-expiration experience. Note the 'renew' call to action and some kind of notice in this mockup (you can ignore any other discrepancies of this view outside of the red bits)

screen shot 2017-08-18 at 11 14 43 am

**Comment by apeatling** [comment]
@sirreal Looks like this one is blocked and it has been a while. If we have more activity or code changes to go, please reopen it.
